File: materias 3 periodo ple/programacao orientada ao objeto/trabalhos praticos/tp-2/SistemaOficina/src/sistemaoficina/Clientes/Carro.java
package sistemaoficina.Clientes;

import java.util.ArrayList;
import java.util.List;

public class Carro {
    private String modelo;
    private String placa;
    private String cor;

    //Construtor
    public Carro(String modelo, String placa, String cor){
        setModelo(modelo);
        setPlaca(placa);
        setCor(cor);
    }

    public Carro() {}

    //Gets

    public String getModelo() {
        return modelo;
    }

    public String getPlaca() {
        return placa;
    }

    public String getCor() {
        return cor;
    } 

    /*
    protected Ordem getListaHistorico(int indice){
        // Percorrer historico
    }
    */

    //Sets

    protected void setModelo(String modelo) {
        this.modelo = modelo;
    }

    protected void setPlaca(String placa) {
        this.placa = placa;
    }

    protected void setCor(String cor) {
        this.cor = cor;
    }
}

File: materias ple/programacao orientada ao objeto/trabalhos praticos/tp-1/Codigo/Cpp/Carro.cpp
#include "../Hpp/Carro.hpp"
 
Carro :: Carro(string modelo, string placa, string cor, int id) : modelo(modelo), placa(placa), cor(cor), id(id){}

Carro :: ~Carro(){};

//Gets

string Carro :: getModelo(){
    return modelo;
}

string Carro :: getPlaca(){
    return placa;
}

string Carro :: getCor(){
    return cor;
}

Ordem Carro :: getListaHistorico(int indice){
    list<Ordem>::iterator i = this->historico.begin();

    advance(i, indice - 1);

    return *i;
}

//Sets

void Carro :: setModelo(string modelo){
    this->modelo = modelo;
}

void Carro :: setPlaca(string placa){
    this->placa = placa;
}

void Carro :: setCor(string cor){
    this->cor = cor;
}




